pgjdbc/src/main/java/org/postgresql/util/PGServiceFile.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.text.MessageFormat;
+import java.util.HashMap;
+
+/*
+ * See https://www.postgresql.org/docs/current/static/libpq-pgservice.html.
+ */
+public class PGServiceFile {
+  public static HashMap load(String service) throws Exception {
+    String filename = findPath();
+    PGServiceFile file = new PGServiceFile();
+    BufferedReader in = new BufferedReader(new FileReader(filename));
+    file.parse(in);
+    return file.getService(service);
+  }
+
+  private static String findPath() {
+    String filename = System.getProperty("postgres.pgservicefile");
+    if (filename == null) {
+      filename = System.getenv().get("PGSERVICEFILE");
+    }
+    if (filename == null) {
+      filename = System.getProperty("user.home") + File.separator + ".pgservices.conf";
+    }
+    return filename;
+  }
+
+  private HashMap<String, HashMap<String, String>> sections;
+
+  public PGServiceFile() {
+    sections = new HashMap();
+  }
+
+  public void parse(BufferedReader in) throws Exception {
```

Comment:
@davecramer I implement the parse method I/O free to ease unit testing this method too.